### PR TITLE
validate that cv warning to force promote

### PR DESCRIPTION
**Pull request by vijaysawant**
### Problem Statement
CU Case - Test cv promote shows warning of 'force promotion' for out of sequence LCE
BZ 2125728

### Solution
Added automation coverage

### Related Issues
NA

### CODE REVIEW

File: tests/foreman/cli/test_contentview.py [hunk 1]
```diff
@@ -4065,6 +4065,62 @@ def test_version_info_by_lce(self, module_org, module_target_sat):
...
+            4. Publish version 2.0
+            5. Try to promote 2.0 on to Preprod
+
+        :expectedresults:
+            1. After step 4, version 1.0 in 'Test' and 2.0 in 'Library'
+            2. After step 5, we will get warning, 'Cannot promote environment out of sequence. Use force to bypass restriction'
+
+        :CaseImportance: High
+
+        :BZ: 2125728
```
Comment (vsedmik) [inline tests/foreman/cli/test_contentview.py:4086]:
```suggestion
        :BZ: 2125728
        
        :customerscenario: true
```

**Review by vsedmik — CHANGES_REQUESTED** [review]
One small change plus there would be some other proposal for change, which would be better discussed live at meet.

File: tests/foreman/cli/test_contentview.py [hunk 2]
```diff
@@ -4065,6 +4065,63 @@ def test_version_info_by_lce(self, module_org, module_target_sat):
...
+
+        :customerscenario: true
+        """
+        # 1. Create an environment path ex- Library >> Test >> Preprod >> Prod
```
Comment (lhellebr) [inline tests/foreman/cli/test_contentview.py:4090]: Should all this entity creation be turned into a fixture?

**Review by lhellebr — APPROVED** [review]

**Comment by lhellebr** [comment]
What Vladimir is suggesting sounds like a very good test proposal but I'd leave it up to you if it is or is not out of scope of this PR.

But more importantly, I doubt this should be marked with the `:BZ:` in docstring. Because I think this tests something else than what the BZ is about. Its title is: `Content view Promote always warns for 'Force promotion' even if correct path is followed. `

**Review by vsedmik — APPROVED** [review]
Just a few nitpicks, non-blocking.

**Review by sambible — APPROVED** [review]
PRT and lgtm

**Review by damoore044 — APPROVED** [review]
nice work! pending @vsedmik 's comments on docstrings

**Review by ColeHiggins2 — APPROVED** [review]

File: tests/foreman/cli/test_contentview.py [hunk 3]
```diff
@@ -4065,6 +4065,74 @@ def test_version_info_by_lce(self, module_org, module_target_sat):
...
+        )
+        lce_prod = target_sat.cli_factory.make_lifecycle_environment(
+            {'organization-id': function_org.id, 'prior': lce_preprod['name']}
+        )
+
+        # Create a CV and publish into the Library
+        cv = target_sat.cli_factory.make_content_view({'organization-id': function_org.id})
+        target_sat.cli.ContentView.publish({'id': cv['id']})
+
+        # Promote version 1.0 to Preprod, skip Library, this should fail with warning
```
[the remark this answers is not in the document]
Comment (vijaysawant) [reply tests/foreman/cli/test_contentview.py:4104]: Can you please update comment with 'suggest changes' so I can commit it directly, (I am away from keyb)

[the remark this answers is not in the document]
Comment (vsedmik) [reply tests/foreman/cli/test_contentview.py:4104]:
```suggestion
        # Promote version 1.0 to Preprod, skip Test, this should fail with warning
```

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_contentview.py -k validate_force_promote_warning

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6105
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_contentview.py -k validate_force_promote_warning --external-logging
Test Result : ========== 1 passed, 131 deselected, 1 warning in 1149.81s (0:19:09) ===========
```

**Pull request merged**